client/mysql_upgrade.c

```diff
@@ -1401,6 +1410,114 @@ static int check_version_match(void)
 }
 
 
+/*
+  Remove .frm archives from directory
+
+  SYNOPSIS
+    dirp      list of files in archive directory
+    db        data base name
+    org_path  path of archive directory
+
+  RETURN
+    > 0 number of removed files
+    -1  error
+
+  NOTE
+    A support of "arc" directories is obsolete, however this
+    function should exist to remove existent "arc" directories.
+*/
+long mysql_rm_arc_files(MY_DIR *dirp, const char *org_path)
+{
+  long deleted= 0;
+  ulong found_other_files= 0;
+  char filePath[FN_REFLEN];
+  DBUG_ENTER("mysql_rm_arc_files");
+  DBUG_PRINT("enter", ("path: %s", org_path));
+
+  for (size_t i=0; i < dirp->number_of_files; i++)
+  {
+    FILEINFO *file=dirp->dir_entry+i;
+    char *extension, *revision;
+    DBUG_PRINT("info",("Examining: %s", file->name));
+
+    extension= fn_ext(file->name);
+    if (extension[0] != '.' ||
+        extension[1] != 'f' || extension[2] != 'r' ||
+        extension[3] != 'm' || extension[4] != '-')
+    {
+      found_other_files++;
+      continue;
+    }
+    revision= extension+5;
+    while (*revision && my_isdigit(system_charset_info, *revision))
+      revision++;
+    if (*revision)
+    {
+      found_other_files++;
+      continue;
+    }
+    strxmov(filePath, org_path, "/", file->name, NullS);
+    if (mysql_file_delete_with_symlink(key_file_misc, filePath, "", MYF(MY_WME)))
```

Comment:
`my_delete` is probably sufficient.  `key_file_misc` is related to peformance schema and is really only for within the server.